README.md
# https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-
# https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-
# https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-
# https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-
# https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-
# https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-
# https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-


history.txt
    1  cd mysite
    2  git status
    3  git add .
    4  git commit -m "ca vous donne donc un huitieme point"
    5  mkvirtualenv flask-tutorial --python=python3.6
    6  pip install flask flask-login flask-sqlalchemy mysql-connector-python
    7  pip freeze > requirements.txt
    8  git status
    9  git add .
   10  git commit -m "quels sont les deux grandes luminaires de ce chapitre"
   11  pip install Flask-Migrate
   12  export FLASK_APP=flask_app.py
   13  flask db init
   14  flask db migrate
   15  flask db stamp head
   16  pip freeze > requirements.txt
   17  git status
   18  git add .
   19  git commit -m "l'hymne ave verum, bien connue par les melomanes"
   20  flask db migrate
   21  flask db upgrade
   22  git status
   23  git add .
   24  git commit -m "place d'italie, j'y vais une fois par mois"
   25  flask db migrate
   26  flask db upgrade
   27  flask shell
   28  select * from users;
   29  echo "# https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-" >> README.md
   30  git init
   31  git add README.md
   32  git commit -m "first commit"
   33  git remote add origin https://github.com/benoitsevres/https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-.git
   34  git push -u origin master
   35  bserv3425@gmail.com
   36  echo "# https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-" >> README.md
   37  git init
   38  git add README.md
   39  git commit -m "first commit"
   40  git remote add origin https://github.com/benoitsevres/https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-.git
   41  git push -u origin master
   42  echo "# https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-" >> README.md
   43  git init
   44  git add README.md
   45  git commit -m "first commit"
   46  git remote add origin https://github.com/benoitsevres/https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-.git
   47  git push -u origin master
   48  echo "# https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-" >> README.md
   49  git init
   50  git add README.md
   51  git commit -m "first commit"
   52  git remote add origin https://github.com/benoitsevres/https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-.git
   53  git push -u origin master
   54  echo "# https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-" >> README.md
   55  git init
   56  git add README.md
   57  git commit -m "first commit"
   58  git remote add origin https://github.com/benoitsevres/https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-.git
   59  git push -u origin master
   60  echo "# https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-" >> README.md
   61  git init
   62  git add README.md
   63  git commit -m "first commit"
   64  git remote add origin https://github.com/benoitsevres/https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-.git
   65  git push -u origin master
   66  echo "# https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-" >> README.md
   67  git init
   68  git add README.md
   69  git commit -m "first commit"
   70  git remote add origin https://github.com/benoitsevres/https-www.pythonanywhere.com-user-danzidascenzi-consoles-9947030-.git
   71  git push -u origin master
   72  history
   73  history > history.txt


flask_app.py
from datetime import datetime
from flask import Flask, redirect, render_template, request, url_for
from flask_login import current_user, login_required, login_user, LoginManager, logout_user, UserMixin
from flask_migrate import Migrate
from flask_sqlalchemy import SQLAlchemy
from werkzeug.security import check_password_hash


app = Flask(__name__)
app.config["DEBUG"] = True

SQLALCHEMY_DATABASE_URI = "mysql+mysqlconnector://{username}:{password}@{hostname}/{databasename}".format(
    username="danzidascenzi",
    password="Exeter22",
    hostname="danzidascenzi.mysql.pythonanywhere-services.com",
    databasename="danzidascenzi$comments",
)
app.config["SQLALCHEMY_DATABASE_URI"] = SQLALCHEMY_DATABASE_URI
app.config["SQLALCHEMY_POOL_RECYCLE"] = 299
app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
db = SQLAlchemy(app)
migrate = Migrate(app, db)

app.secret_key = "46546546546546546546"
login_manager = LoginManager()
login_manager.init_app(app)


class User(UserMixin, db.Model):

    __tablename__ = "users"

    id = db.Column(db.Integer, primary_key=True)
    username = db.Column(db.String(128))
    password_hash = db.Column(db.String(128))

    def check_password(self, password):
        return check_password_hash(self.password_hash, password)


    def get_id(self):
        return self.username


@login_manager.user_loader
def load_user(user_id):
    return User.query.filter_by(username=user_id).first()


class Comment(db.Model):

    __tablename__ = "comments"

    id = db.Column(db.Integer, primary_key=True)
    content = db.Column(db.String(4096))
    posted = db.Column(db.DateTime, default=datetime.now)
    commenter_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=True)
    commenter = db.relationship('User', foreign_keys=commenter_id)


@app.route("/", methods=["GET", "POST"])
def index():
    if request.method == "GET":
        return render_template("main_page.html", comments=Comment.query.all())

    if not current_user.is_authenticated:
        return redirect(url_for('index'))

    comment = Comment(content=request.form["contents"], commenter=current_user)
    db.session.add(comment)
    db.session.commit()
    return redirect(url_for('index'))



@app.route("/login/", methods=["GET", "POST"])
def login():
    if request.method == "GET":
        return render_template("login_page.html", error=False)

    user = load_user(request.form["username"])
    if user is None:
        return render_template("login_page.html", error=True)

    if not user.check_password(request.form["password"]):
        return render_template("login_page.html", error=True)

    login_user(user)
    return redirect(url_for('index'))



@app.route("/logout/")
@login_required
def logout():
    logout_user()
    return redirect(url_for('index'))

migrations/versions/b2f5455415cc_.py
"""empty message

Revision ID: b2f5455415cc
Revises: 
Create Date: 2018-08-23 22:43:31.306976

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = 'b2f5455415cc'
down_revision = None
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('users',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('username', sa.String(length=128), nullable=True),
    sa.Column('password_hash', sa.String(length=128), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.create_table('comments',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('content', sa.String(length=4096), nullable=True),
    sa.Column('posted', sa.DateTime(), nullable=True),
    sa.Column('commenter_id', sa.Integer(), nullable=True),
    sa.ForeignKeyConstraint(['commenter_id'], ['users.id'], ),
    sa.PrimaryKeyConstraint('id')
    )
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_table('comments')
    op.drop_table('users')
    # ### end Alembic commands ###


migrations/versions/bb1ad5fa6892_.py
"""empty message

Revision ID: bb1ad5fa6892
Revises: b2f5455415cc
Create Date: 2018-08-23 22:49:06.841490

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = 'bb1ad5fa6892'
down_revision = 'b2f5455415cc'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('users',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('username', sa.String(length=128), nullable=True),
    sa.Column('password_hash', sa.String(length=128), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.add_column('comments', sa.Column('commenter_id', sa.Integer(), nullable=True))
    op.add_column('comments', sa.Column('posted', sa.DateTime(), nullable=True))
    op.create_foreign_key(None, 'comments', 'users', ['commenter_id'], ['id'])
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_constraint(None, 'comments', type_='foreignkey')
    op.drop_column('comments', 'posted')
    op.drop_column('comments', 'commenter_id')
    op.drop_table('users')
    # ### end Alembic commands ###
